src/store.rs

```diff
@@ -9,39 +10,31 @@ pub struct Store {
 }
 
 impl Store {
-    pub fn new(url: &str, expire_keys: u64) -> Result<Store, &'static str> {
-        let res = redis::Client::open(url);
-        if res.is_err() {
-            return Err("error opening store connection");
-        }
-        Ok(Store { client: res.unwrap(), expire_keys: expire_keys })
+    pub fn new(url: &str, expire_keys: u64)
+               -> Result<Store, String> {
+        let client = try!(redis::Client::open(url).map_err(stringify_err));
+        Ok(Store { client: client, expire_keys: expire_keys })
     }
 
-    pub fn store_session(&self, key: &str, data: &[(&str, &str)])
+    pub fn store_session(&self, session_id: &str, data: &[(&str, &str)])
                          -> Result<(), String> {
-        let res: RedisResult<String> = self.client.hset_multiple(key, data);
-        if res.is_err() {
-            return Err(res.unwrap_err().to_string());
-        }
-        let res: RedisResult<bool> =
-            self.client.expire(key, self.expire_keys as usize);
-        if res.is_err() {
-            return Err(res.unwrap_err().to_string());
-        }
+        let key = format_session_key(session_id);
+        try!(self.client.hset_multiple(key.as_str(), data).map_err(stringify_err));
+        try!(self.client.expire(key.as_str(), self.expire_keys as usize).map_err(stringify_err));
         Ok(())
     }
 
-    pub fn get_session(&self, key: &str)
+    pub fn get_session(&self, session_id: &str)
                        -> Result<HashMap<String, String>, String> {
-        let res: RedisResult<HashMap<String, String>> =
-            self.client.hgetall(key);
-        if res.is_err() {
-            return Err(res.unwrap_err().to_string());
-        }
-        let stored = res.unwrap();
-        if stored.is_empty() {
-            return Err("session not found".to_string());
-        }
-        Ok(stored)
+        let key = format_session_key(session_id);
+        self.client.hgetall(key.as_str()).map_err(stringify_err)
     }
 }
+
+fn stringify_err(err: RedisError) -> String {
```

Comment:
Yeah, I guess then implementing our own error type is probably the right solution. I'm thinking it would be an `enum` covering all the input error types (e.g. start with `RedisError`), then we can implement `From` for `RedisError` and `Into` for `IronError`.